webrtc-c/canary/src/CanaryWebrtc.cpp

```diff
@@ -150,6 +153,44 @@ STATUS run(Canary::PConfig pConfig)
     return retStatus;
 }
 
+VOID runPeer(Canary::PConfig pConfig, TIMER_QUEUE_HANDLE timerQueueHandle, STATUS* pRetStatus)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT32 timeoutTimerId;
+
+    Canary::Peer::Callbacks callbacks;
+    callbacks.onNewConnection = onNewConnection;
+    callbacks.onDisconnected = []() { terminated = TRUE; };
+
+    Canary::Peer peer;
+
+    CHK(pConfig != NULL, STATUS_NULL_ARG);
+
+    pConfig->print();
+    CHK_STATUS(peer.init(pConfig, callbacks));
+    CHK_STATUS(peer.connect());
+
+    {
+        // Since the goal of the canary is to test robustness of the SDK, there is not an immediate need
+        // to send audio frames as well. It can always be added in if needed in the future
+        std::thread videoThread(sendCustomFrames, &peer, MEDIA_STREAM_TRACK_KIND_VIDEO, pConfig->bitRate.value, pConfig->frameRate.value);
+        // All metrics tracking will happen on a time queue to simplify handling periodicity
+        CHK_STATUS(timerQueueAddTimer(timerQueueHandle, METRICS_INVOCATION_PERIOD, METRICS_INVOCATION_PERIOD, canaryRtpOutboundStats, (UINT64) &peer,
```

Comment:
Just a note: Alternatively, the handler could simply report all 3 metrics when invoked. However, this works too.

Reply:
Ah that makes sense as well! I'll keep things as they are for now. We can iterate more later when needed.